### PR TITLE
Issue pro 773

### PR DESCRIPTION
**Pull request by jdicks**

### CODE REVIEW

**Comment by jdicks** [comment]
This is the test project (including c domain statechart) i majorly used
[Test_BigProject_pro773.zip](https://github.com/Yakindu/statecharts/files/1127143/Test_BigProject_pro773.zip)

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramEditor.java [hunk 1]
```diff
@@ -329,4 +333,33 @@ public Object getAdapter(@SuppressWarnings("rawtypes") Class type) {
...
+	private CrossReferenceAdapter getCrossReferenceAdapter(final TransactionalEditingDomain domain) {
+
+		final EList<Adapter> eAdapters = domain.getResourceSet().eAdapters();
+		for (final Adapter adapter : eAdapters) {
+			if (adapter instanceof CrossReferenceAdapter) {
+				return (CrossReferenceAdapter) adapter;
+			}
+		}
+		return null;
+	}
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramEditor.java]: formatting

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/ShadowModelValidationJob.java [hunk 2]
```diff
@@ -85,6 +88,10 @@ public boolean isCanceled() {
 
 	protected void cloneResource(final IProgressMonitor monitor, final Resource shadowResource)
 			throws ExecutionException {
+		// clear contents before contents are copied...
+		shadowResource.getContents().clear();
+//		shadowResource.eAdapters().clear();
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/ShadowModelValidationJob.java]: valid comment or can be removed?

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramEditor.java [hunk 3]
```diff
@@ -329,4 +333,33 @@ public Object getAdapter(@SuppressWarnings("rawtypes") Class type) {
...
+	    }
+
+	private CrossReferenceAdapter getCrossReferenceAdapter(final TransactionalEditingDomain domain) {
+
+		final EList<Adapter> eAdapters = domain.getResourceSet().eAdapters();
+		for (final Adapter adapter : eAdapters) {
+			if (adapter instanceof CrossReferenceAdapter) {
+				return (CrossReferenceAdapter) adapter;
+			}
+		}
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramEditor.java]: Isn't there any getAdapter(Class) method already? Not sure though..

Comment (jdicks) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramEditor.java]: no :(

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/ShadowModelValidationJob.java (hunk 2 above)
Comment (jdicks) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/ShadowModelValidationJob.java]: can be removed...

**Pull request closed without merging**